Rewrite vizsettings in normal react code

syt123450 opened the pull request:
As vizsettings was originally plain javascript, in this PR, I rewrite part of vizsettings in a more "normal react" way. The changes I did are listed below:

1. Rename React class to uppercase(the original lower case can not be used as &lt;XXX /&gt;)
2. Write new React classes to replace original static dom functions
3. Replace document.createElement() with native react class tag
4. Move util function valToStr to underscore_ext.js
5. Organize all utils functions in a inner object settingUtils (In this way, it will be easier for us to organize these kind of logics in future, not scattered in different parts)
6. Delete useless img import

Looking forward for your review and suggestions. @acthp 

Thanks.

acthp commented on js/views/VizSettings.js:
```diff
@@ -47,55 +46,66 @@ import vizSettingStyle from "./VizSettings.module.css";
+					{isAllFloat ? <AllFloat /> : null}
+					{isSV ? <Sv /> : null}
+					{noSetting ? <NoSettings /> : null}
+					<FinishButtonBar />
```
Since these elements are mutually exclusive, it might be clearer to write it that way, like
```
var Settings = valueType === "float" || valueType === 'segmented' ? <AllFloat /> :
               valueType === "mutation" && fieldType === 'SV' ? <Sv /> :
               <NoSettings />;
```
then render ```<Settings />```

 For nested ternary expressions, like I'm suggesting, I recommend formatting like

```
    predicate ? value :
    predicate ? value :
    predicate ? value :
    defaultValue
```

syt123450 replied:
I use settingContent parameter to replace the previous code.

acthp commented on js/views/VizSettings.js:
```diff
@@ -120,71 +130,72 @@ function vizSettingsWidget(node, onVizSettings, vizState, id, hide, defaultNorma
 		}
 	}
+	var settingUtils = {
+		getInputSettingsFloat: function(settings) {
```
I don't think it's worth collecting helper functions like this, within a module, unless it serves some functional purpose, like dispatch on a key. It might make sense to move them to the top of the file, or to the top of whatever section of the file they are used in. Some of these are going to be deleted, though, when all the raw DOM code is gone, because we can use a simpler state model via React. So, probably not worth changing now.

I recommend dropping the settingUtils changes.

syt123450 replied:
I have dropped the settingsUitls changes.

syt123450 commented:
@acthp Thank you for your review and suggestions. I have changed the code based on your suggestions. To make the commit history clean, I have squashed all commits into one.
